### ISSUE TITLE
Not able to write_videofile()

### ISSUE THREAD

**Issue by MohnishMagesh** [Zulko/moviepy#1183]
I am trying to create a composite video clip (in a for loop) with four imageClips like so
- I then append these imageClips to a list with each iteration of the for loop
- Then I put this as a list to the compositeVideoClip
When I try and use method write_videofile(), that's when the error occurs.
I've tried using a static path but the result is the exact same.

```
path_defined_for_clips = SOME_PATH
comment_clips = []
for j in range(1,5):
    pid = f"{path_defined_for_clips}\\folder"
    path_of_sub_audio = f"{pid}\\sub_{j}.mp3"
    path_of_sub_image = f"{pid}\\sub_{j}.png"
    audio = AudioFileClip(path_of_sub_audio)
    # image = ImageClip(path_of_sub_image).set_duration(audio.duration).set_fps(5).set_audio(audio).set_position(("center","top"))
    image = ImageClip(path_of_sub_image).set_duration(audio.duration).set_fps(5).set_audio(audio).set_position((0.0,0.2),relative=True)
    # image.with_duration(audio.duration).with_position(("center","top"))
    image = image.resize(1.875)
    final_clip = CompositeVideoClip([image],size=(1920,1080),bg_color=(0,0,255))
    # final_clip.write_videofile("C:\\Users\\MOHNISH\\AI\\Reddit_bot\\movie1.mp4")
    comment_clips.append(final_clip)

comment_video_clip_for_now = concatenate_videoclips(comment_clips,method="compose")

comment_video_clip_for_now.write_videofile("movie1.mp4")
```

and this is what the error looks like

```
Moviepy - Building video movie1.mp4.
MoviePy - Writing audio in movie1TEMP_MPY_wvf_snd.mp3
chunk:   0%|          | 0/276 [00:00<?, ?it/s, now=None]
chunk:   1%|          | 2/276 [00:00<00:35,  7.75it/s, now=None]
chunk:  10%|#         | 28/276 [00:00<00:23, 10.46it/s, now=None]
chunk:  24%|##3       | 66/276 [00:00<00:14, 14.76it/s, now=None]
chunk:  38%|###8      | 105/276 [00:00<00:08, 20.75it/s, now=None]
chunk:  52%|#####2    | 144/276 [00:01<00:04, 28.97it/s, now=None]
chunk:  65%|######4   | 179/276 [00:01<00:02, 39.91it/s, now=None]Traceback (most recent call last):
  File "C:\Users\MOHNISH\AI\Project_personal\tts_watson.py", line 35, in <module>
    comment_video_clip_for_now.write_videofile("movie1.mp4")
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-55>", line 2, in write_videofile
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-54>", line 2, in write_videofile
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 135, in use_clip_fps_by_default
    return f(clip, *new_a, **new_kw)
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-53>", line 2, in write_videofile
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 22, in convert_masks_to_RGB
    return f(clip, *a, **k)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\video\VideoClip.py", line 298, in write_videofile
    logger=logger)
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-45>", line 2, in write_audiofile
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 210, in write_audiofile
    logger=logger)
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-9>", line 2, in ffmpeg_audiowrite
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\io\ffmpeg_audiowriter.py", line 169, in ffmpeg_audiowrite
    logger=logger):
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 86, in iter_chunks
    fps=fps, buffersize=chunksize)
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-44>", line 2, in to_soundarray
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 127, in to_soundarray
    snd_array = self.get_frame(tt)
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-11>", line 2, in get_frame
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\Clip.py", line 93, in get_frame
    return self.make_frame(t)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 297, in make_frame
    for c, part in zip(self.clips, played_parts)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 298, in <listcomp>
    if (part is not False)]
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-11>", line 2, in get_frame
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\Clip.py", line 93, in get_frame
    return self.make_frame(t)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 297, in make_frame
    for c, part in zip(self.clips, played_parts)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\AudioClip.py", line 298, in <listcomp>
    if (part is not False)]
  File "<C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\decorator.py:decorator-gen-11>", line 2, in get_frame
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\Clip.py", line 93, in get_frame
    return self.make_frame(t)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\io\AudioFileClip.py", line 77, in <lambda>
    self.make_frame = lambda t: self.reader.get_frame(t)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\io\readers.py", line 185, in get_frame
    self.buffer_around(fr_max)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\io\readers.py", line 239, in buffer_around
    array = self.read_chunk(chunksize)
  File "C:\Users\MOHNISH\AppData\Local\Programs\Python\Python37-32\lib\site-packages\moviepy\audio\io\readers.py", line 113, in read_chunk
    s = self.proc.stdout.read(L)
AttributeError: 'NoneType' object has no attribute 'stdout'
```

I'm not exactly sure what is it that I've done wrong here.

**Comment by tburrows13** [comment]
Hi, this error isn't your fault, and there's a few known workarounds. The simplest way is to install moviepy v1.0.0 with `pip install moviepy=1.0.0`. Alternatively, go to `moviepy/audio/io/AudioFileClip.py` and delete the line `self.reader.close_proc()`, as seen in #1059. Note that this stops it working on Windows for Python versions below 3.6.

This will be fixed in the next MoviePy update (1.0.3 or 2.0.0).

**Comment by tburrows13** [comment]
Update: you can get the fix now by updating moviepy to 1.0.3.
